controllers/profile.go

```diff
@@ -0,0 +1,52 @@
+package controllers
+
+import (
+	"net/http"
+
+	"google.golang.org/appengine/datastore"
+
+	"encoding/json"
+
+	"github.com/coduno/app/model"
+	"github.com/coduno/engine/util"
+	"github.com/gorilla/mux"
+	"golang.org/x/net/context"
+)
+
+func GetProfileByKey(ctx context.Context, w http.ResponseWriter, r *http.Request) (int, error) {
+	if err := util.CheckMethod(r, "GET"); err != nil {
+		return http.StatusMethodNotAllowed, err
+	}
+
+	key, err := datastore.DecodeKey(mux.Vars(r)["key"])
+
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	var profile model.Profile
+	err = datastore.Get(ctx, key, &profile)
+	if err != nil {
```

Comment:
This could be:

```go
var profile model.Profile
if err := datastore.Get(ctx, key, &profile); err != nil {
	return http.StatusInternalServerError, err
}
```